docs: add styling page to the AsyncAPI style guide (#1970)
asyncapi/community: docs/011-styleguide/styling.md

---
title: Styling

> description: This style guide provides styling requirements for all visual and written content in AsyncAPI documentation. This includes CSS code, fontsize, markdown, lists, images, tables, diagrams, and so on.

Reviewer: ```suggestion
description: This style guide provides styling requirements for all visual and written content in AsyncAPI documentation. This includes CSS code, font size, Markdown, lists, images, tables, diagrams, and so on.
```

weight: 150
---

# Styling Guidelines

## CSS Code Style

> ### CSS Naming Convention

Reviewer 2: > can you clarify more on this

In [my suggestion](https://github.com/asyncapi/community/pull/1970#discussion_r2187026656), I introduced several new definitions that haven't been mentioned before, like _rule_ and _declaration_. So they need to be described in this section to bring clarity and consistency to naming convention and avoid potential confusion in the future when referencing certain css 'building blocks'.

## Typography
Consistent typography is key to readability and visual hierarchy.

### Font
- **Primary Font:** Inter (or a fallback sans-serif like Helvetica, Arial).

- **Monospace Font:** Roboto Mono for all code blocks and inline code.

### Font Size & Weight
- **Base Font Size:** 16px for body text.

- **Line Height:** 1.6 for body text to ensure readability.

| Element | Font Size | Font Weight | Notes  |
|--------|-----|------------|------------|
| `<h1>` | `2.5rem` | `700`(Bold) | Page Title, one per page. |
| `<h2>` | `2rem` | `700`(Bold) | Major Section Headings.    |
| `<h3>` | `1.75rem` | `600`(Semi-Bold) | Sub-section Headings.     |
| `<h4>` | `1.5rem` | `600`(Semi-Bold) | Smaller sub-section Headings.     |
| Body/`<p>` | `1rem`(16px) | `400`(Regular) | Body of the page.     |
| Inline Code ` ` | `0.9rem` | `400`(Regular) | Use the monospace font.     |
| `Caption/Small` | `0.875rem` | `400`(Regular) | For image captions or helper text.    |

## Markdown
Standardize Markdown usage for consistency across all documents.

- **Emphasis:** Use asterisks for italics (*italic*) and bold (**bold**).

> - **Links:** Always use descriptive link text. Avoid "click here."

Author: I gave an example for links in the following line, for the reference style links the example you gave is different from the link you shared. please clarify more on it

   - **✅Good:** See our [Contribution Guide](/docs/community/contribution-guide) for more.

   - **❌Bad:** For more info, [click here](https://.../contribution-guide).

- **Code Blocks:** Use backticks for inline code (`const foo = 'bar';`). For code blocks, use triple backticks and specify the language for syntax highlighting.

```yaml
asyncapi: 2.5.0
info:
  title: Streetlights API
  version: '1.0.0'
```

> ## Lists

Reviewer 2: The previous revision had a strange and confusing structure:
- Markdown section with emphasis, links, and code blocks as bullet points
  - Emphasis
  - Links
  - Code Blocks
- Lists (being another Markdown element, but listed as a separate section)
- Images (being another Markdown element, but listed as a separate section)
- Diagrams
- Tables (being another Markdown element, but listed as a separate section)

Now, it looks more accurate even though I think Images can be moved under Markdown section as well:
- Markdown
  - Emphasis
  - Links
  - Code Blocks
  - Lists
  - Tables
- Images
- Diagrams

> ## Lists

Author: I have updated the document. Kindly review

Diagrams are crucial for explaining complex architectures. Consistency is key.

- **Tool:** Whenever possible, use **Mermaid.js** to create diagrams directly within Markdown. This keeps them version-controlled and easy to update.

- **Style:**

  - **Colors:** Adhere to the AsyncAPI color palette. Use colors meaningfully to distinguish between different types of components (e.g., service, message, topic).

  - **Flow:** Diagrams should flow logically, typically from left-to-right or top-to-bottom.

  - **Clarity:** Avoid clutter. Use clear, concise labels. If a diagram becomes too complex, break it into multiple, simpler diagrams.

```mermaid
graph TD;
    A[Client] -->|Sends Message| B(Message Broker);
    B -->|Forwards to| C{Application};
    C -->|Processes & Responds| B;
```

## Tables
Use Markdown's pipe table format for simplicity. Ensure columns are properly aligned for readability in the raw markdown file.

| Feature        | Status      | Assigned To |
|----------------|-------------|-------------|
| New Parser     | In Progress | @jane.doe   |
| Kafka Binding  | Done        | @john.doe   |
| Docs Overhaul  | Blocked     | @team/docs  |

## Additional Resources

> - https://getbem.com/introduction/

Reviewer: ```suggestion
- [BEM](https://getbem.com/introduction/)
```

> - https://m2.material.io/design/typography/understanding-typography.html

Reviewer: ```suggestion
- [Understanding Typography](https://m2.material.io/design/typography/understanding-typography.html)
```

> - https://www.markdownguide.org/basic-syntax/

Reviewer: ```suggestion
- [Markdown Guide](https://www.markdownguide.org/basic-syntax/)
```

| Element | Font Size | Font Weight | Notes  |
|--------|-----|------------|------------|
| `<h1>` | `2.5rem` | `700`(Bold) | Page title, one per page. |
| `<h2>` | `2rem` | `700`(Bold) | Major section headings.    |
| `<h3>` | `1.75rem` | `600`(Semi-Bold) | Sub-section headings.     |
| `<h4>` | `1.5rem` | `600`(Semi-Bold) | Smaller sub-section Headings.     |
| Body/`<p>` | `1rem`(16px) | `400`(Regular) | Body text.     |
| Inline code (`code`)  | `0.9rem` | `400`(Regular) | Uses the monospace font.     |
| `Caption` / small | `0.875rem` | `400`(Regular) | Image captions or helper text.    |

## Markdown
Standardize Markdown usage for consistency across all documents.

### Emphasis
Use asterisks for italics (`*italic*`) and bold (`**bold**`). You can also use underscores (`_underscores_`) for emphasis.

### Links
Use square brackets to describe links and a parenthesis to insert the actual links. Always use descriptive link text. Avoid phrases like "click here."

   - **✅ Good:** See our [Contribution Guide](/docs/community/contribution-guide) for more.

   - **❌ Bad:** For more info, [click here](https://.../contribution-guide).

> 

Reviewer: @ezinneanne this suggestion is worth looking into
